src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateImplicitNamespaceImports.targets

```diff
@@ -0,0 +1,68 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.GenerateImplicitNamespaceImports.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    ============================================================
+                                     GenerateImplicitNamespaceImports
+
+    Generates implicit namespace imports source to intermediate directory for C# projects
+    ============================================================
+    -->
+  <PropertyGroup>
+    <GeneratedImplicitNamespaceImportFile Condition="'$(GeneratedImplicitNamespaceImportFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).ImplicitNamespaceImports$(DefaultLanguageSourceExtension)</GeneratedImplicitNamespaceImportFile>
+    <DisableImplicitNamespaceImports_DotNet Condition="'$(DisableImplicitNamespaceImports_DotNet)' == '' and '$(DisableImplicitNamespaceImports)' != 'true'">false</DisableImplicitNamespaceImports_DotNet>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(DisableImplicitNamespaceImports_DotNet)' != 'true'
```

Comment:
Can we put this `ItemGroup` in a `.props` file instead of a `.targets` file?  That way it's simpler to remove an `Import` you don't want in the body of a project file:

```xml
<ItemGroup>
    <Import Remove="System.Threading.Tasks"/>
</ItemGroup>
```

This works because of MSBuilds multi-pass evaluation.  Properties are evaluated in the first pass, so you can put items in `.props` files that have conditions based on properties that are set "later".

I think we should also move the [VB import definitions](https://github.com/dotnet/sdk/blob/cfc3463578c94dba51115d8fc218a4a7d96281c2/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.VisualBasic.targets#L16-L39) from Microsoft.NET.Sdk.VisualBasic.targets to the same .props file.

Reply:
I do not think we should move the VB Import definitions this late in .NET 6, unless it is super evident that it cannot break anything.

Reply:
>  Properties are evaluated in the first pass, so you can put items in .props files that have conditions based on properties that are set "later".

@dsplaisted I think I'm missing something here, how does this work? I originally placed the Import items in the .target files since they needed to be TFM driven. Seems like you are suggesting that it's possible to do that even if the itemgroup is defined in .props files and I'd like to make sure I understand how that works before I make the change.

Reply:
Oh I think I understand what you mean, let me test a few things and I'll make the change if it works.

Reply:
Yeah, this is the way the default globs work.  See Microsoft.NET.Sdk.DefaultItems.props.